### ISSUE TITLE
SLF4J: Class path contains multiple SLF4J bindings.

### ISSUE THREAD

**Issue by Postremus** [quarkusio/quarkus#7968]
**Describe the bug**
I get warnings about multiple SLF4J bindings, when I try to use log4j1 with slf4j for application logging.

**Expected behavior**
No warning about multiple log bindings.

**Actual behavior**
I get the following warning. I can just ignore it, but it is scary.
It should probably be investigated.

SLF4j picks the first (I think) binding implementation on the classpath.
Therefore, if my dependencies change, logging would not work anymore.

```
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/C:/Users/mnp/.m2/repository/org/jboss/slf4j/slf4j-jboss-logging/1.2.0.Final/slf4j-jboss-logging-1.2.0.Final.jar!/org/slf4j/impl/StaticLog
gerBinder.class]
SLF4J: Found binding in [jar:file:/C:/Users/mnp/.m2/repository/org/slf4j/slf4j-log4j12/1.7.29/slf4j-log4j12-1.7.29.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.jboss.slf4j.JBossLoggerFactory]
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/C:/Users/mnp/.m2/repository/org/slf4j/slf4j-log4j12/1.7.29/slf4j-log4j12-1.7.29.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/C:/Users/mnp/.m2/repository/org/jboss/slf4j/slf4j-jboss-logging/1.2.0.Final/slf4j-jboss-logging-1.2.0.Final.jar!/org/slf4j/impl/StaticLog
gerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/C:/Users/mnp/.m2/repository/org/slf4j/slf4j-log4j12/1.7.29/slf4j-log4j12-1.7.29.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/C:/Users/mnp/.m2/repository/org/jboss/slf4j/slf4j-jboss-logging/1.2.0.Final/slf4j-jboss-logging-1.2.0.Final.jar!/org/slf4j/impl/StaticLog
gerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2020-03-19 12:59:24,664 WARN  [io.qua.agr.dep.AgroalProcessor] (build-4) The Agroal dependency is present but no JDBC datasources have been defined.
2020-03-19 12:59:27,516 WARN  [io.qua.net.run.NettyRecorder] (Thread-24) Localhost lookup took more than one second, you need to add a /etc/hosts entry to improve Quarkus s
tartup time. On Windows the path is C:\Windows\System32\Drivers\etc\hosts. See https://thoeni.io/post/macos-sierra-java/ for macOS details.
2020-03-19 12:59:27,551 INFO  [io.quarkus] (main) code-with-quarkus 1.0.0-SNAPSHOT (powered by Quarkus 1.3.0.Final) started in 4.225s. Listening on: http://0.0.0.0:8080
2020-03-19 12:59:27,552 INFO  [io.quarkus] (main) Profile dev activated. Live Coding activated.
2020-03-19 12:59:27,552 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, liquibase, narayana-jta, resteasy]
Batchvorgang abbrechen (J/N)? j
```

**To Reproduce**
1. Download Reproducer project: 
[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/4354307/code-with-quarkus.zip)
2. Start dev mode
3. See warning in console output

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
MSYS_NT-10.0 NANB7NLNVP2 2.10.0(0.325/5/3) 2018-06-13 23:34 x86_64 Msys
 - Output of `java -version`: 
java version "1.8.0_241"
Java(TM) SE Runtime Environment (build 1.8.0_241-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.241-b07, mixed mode)
 - Quarkus version or git rev: 
1.3.0.Final

 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
Maven home: C:\eclipse\tools\apache-maven\bin\..
Java version: 1.8.0_181, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk1.8.0_181\jre
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

**Comment by oztimpower** [comment]
I'm seeing this also, I fiddled around endlessly trying to solve it and couldn't; I thought it was an issue with my dependencies and gave up.

**Comment by gsmet** [comment]
Isn't the message clear? You have 2 conflicting SLF4J implementations.

Are you only using Quarkus extensions? If so, we need to fix our dependencies.

Just check what `mvn dependency:tree` says and who is responsible for this dependency.

**Comment by geoand** [comment]
I think the question boils down to: `How do I get rid of jboss-logging if I want to use slf4j`

**Comment by Postremus** [comment]
@gsmet Yes, the message is clear, just my explanation isn't. :)

I have as dependencies in my project, quarkus-resteasy, quarkus-liquibase, slf4j, slf4j-log4j12, and log4j.
One implementation comes from slf4j-log4j12.
I guess the other implementation must come from quarkus-resteasy or quarkus-liquibase.

**Comment by gsmet** [comment]
I don't think you can get rid of JBoss Logging.

**Comment by oztimpower** [comment]
In my case I was using the AWS Java SDK v2, along with AWS Java SDK v1 Lambda, which brought in all the above.

**Comment by xmlking** [comment]
in fact there is a bug in 1.5.0 https://github.com/quarkusio/quarkus/issues/9715

**Comment by aomsweet** [comment]
log4j2 is excellent, please support it.

**Comment by Postremus** [comment]
In my project, I fixed this issue as follows:
I configured most of my logging setup following the guide at: https://quarkus.io/guides/logging

To support logging using the log4j SMTPAppender (errors as emails), I added the log4j-jboss-logging as dependency.
I then configured in a `@Startup` bean the log4j root Logger to with an SMTPAppender.
And it just works.

I also do not get this warning anymore, which is a nice bonus.
Closing this, since the quarkus intended way works perfectly for my now.
